spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/test/java/org/springframework/cloud/function/adapter/gcp/FunctionInvokerHttpTests.java

```diff
@@ -52,27 +56,38 @@ public class FunctionInvokerHttpTests {
 
 	@Test
 	public void testHelloWorldSupplier() throws Exception {
-		testHttpFunction(HelloWorldSupplier.class, null, "Hello World!");
+		testHttpFunction(HelloWorldSupplier.class, null, "Hello World!", false, false);
 	}
 
 	@Test
 	public void testJsonInputFunction() throws Exception {
 		testHttpFunction(JsonInputFunction.class, new IncomingRequest("hello"),
-				"Thank you for sending the message: hello");
+				"Thank you for sending the message: hello", false, false);
 	}
 
 	@Test
 	public void testJsonInputOutputFunction() throws Exception {
 		testHttpFunction(JsonInputOutputFunction.class, new IncomingRequest("hello"),
-				new OutgoingResponse("Thank you for sending the message: hello"));
+				new OutgoingResponse("Thank you for sending the message: hello"), false, false);
 	}
 
 	@Test
 	public void testJsonInputConsumer_Background() throws Exception {
-		testHttpFunction(JsonInputConsumer.class, new IncomingRequest("hello"), null);
+		testHttpFunction(JsonInputConsumer.class, new IncomingRequest("hello"), null, false, false);
 	}
 
-	private <I, O> void testHttpFunction(Class<?> configurationClass, I input, O expectedOutput) throws Exception {
+	@Test
+	public void testStatusCodeSet() throws Exception {
+		testHttpFunction(statusCodeSupplier.class, "hello", "hello", true, false);
+	}
+
+	@Test
+	public void testMultiValueHeaderSupplied() throws Exception {
+		testHttpFunction(multiValueHeaderSupplier.class, "hello", "hello", false, true);
+	}
+
+
+	private <I, O> void testHttpFunction(Class<?> configurationClass, I input, O expectedOutput, boolean statusCodeCheck, boolean multiHeaderCheck) throws Exception {
```

Comment:
I think the original test setup gave `testHttpFunction()` helper too much to do.
In the testing structure of  `setup - run - verify`, the specific functionality tested by the test should ideally be contained in the test itself. `testHttpFunction`.

Could you make the mock declarations class members instead of local variables, and handle the mock wiring (when/then setup that's needed by all tests) in a `@BeforeEach` setup method?
Then each individual test should handle creating its own input (which it currently does), executing the functinality (`handler.service(request, response);` and likely closing the buffer, which it currently does not), and, most importantly, its own verification step.
Then you won't need special-case conditionals in the helper method.